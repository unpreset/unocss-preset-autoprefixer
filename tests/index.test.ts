import { createGenerator } from 'unocss'
import presetUno from '@unocss/preset-uno'
import { describe, expect, test } from 'vitest'
import presetPrefixer from '../src'

describe('preset-prefixer', () => {
  const uno = createGenerator({
    presets: [
      presetUno(),
      presetPrefixer(),
    ],
  })

  function generate(input: Parameters<typeof uno.generate>[0]) {
    return uno.generate(input, { preflights: false })
  }

  test('flex autoprefixer', async () => {
    const { css } = await generate(['flex'])

    expect(css).toMatchSnapshot()
  })

  test('text space', async () => {
    const { css } = await generate(['text-1'])

    expect(css).toMatchSnapshot()
  })

  test('font weight', async () => {
    const { css } = await generate(['font-400', 'font-bold'])

    expect(css).toMatchSnapshot()
  })

  test('background gradient', async () => {
    const { css } = await generate(['bg-gradient-to-tr', 'from-purple-500', 'to-pink-500'])

    expect(css).toMatchSnapshot()
  })

  test('p-x', async () => {
    const { css } = await generate(['px-10'])

    expect(css).toMatchSnapshot()
  })
  test('space-x', async () => {
    const { css } = await generate(['space-x-10'])

    expect(css).toMatchSnapshot()
  })
})
