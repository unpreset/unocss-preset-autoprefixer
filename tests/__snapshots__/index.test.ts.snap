// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`preset-prefixer > background gradient 1`] = `
"/* layer: default */
.from-purple-500{--un-gradient-from-position:0%;--un-gradient-from:rgb(168 85 247 / var(--un-from-opacity, 1)) var(--un-gradient-from-position);--un-gradient-to-position:100%;--un-gradient-to:rgb(168 85 247 / 0) var(--un-gradient-to-position);--un-gradient-stops:var(--un-gradient-from), var(--un-gradient-to);}
.to-pink-500{--un-gradient-to-position:100%;--un-gradient-to:rgb(236 72 153 / var(--un-to-opacity, 1)) var(--un-gradient-to-position);}
.bg-gradient-to-tr{--un-gradient-shape:to top right;--un-gradient:var(--un-gradient-shape), var(--un-gradient-stops);background-image:linear-gradient(var(--un-gradient));}"
`;

exports[`preset-prefixer > flex autoprefixer 1`] = `
"/* layer: default */
.flex{display:-webkit-box;display:-webkit-flex;display:flex;}"
`;

exports[`preset-prefixer > font weight 1`] = `
"/* layer: default */
.font-400{font-weight:400;}
.font-bold{font-weight:700;}"
`;

exports[`preset-prefixer > p-x 1`] = `
"/* layer: default */
.px-10{padding-left:2.5rem;padding-right:2.5rem;}"
`;

exports[`preset-prefixer > space-x 1`] = `
"/* layer: default */
.space-x-10>:not([hidden])~:not([hidden]){--un-space-x-reverse:0;margin-left:calc(2.5rem*calc(1 - var(--un-space-x-reverse)));margin-right:calc(2.5rem*var(--un-space-x-reverse));}"
`;

exports[`preset-prefixer > text space 1`] = `
"/* layer: default */
.text-1{font-size:.25rem;}"
`;
